doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -574,56 +568,38 @@ Navigation in the proof tree
    :n:`@natural:`
      Focuses on the :token:`natural`\-th goal to prove.
 
-   :n:`[ @ident ]: %{`
-     Focuses on the named goal :token:`ident`.
+.. _focus_shelved_goal:
 
-   .. note::
+   :n:`[ @ident ]: %{`
+     Focuses on the goal named :token:`ident` even if the goal is not in focus.
+     Goals are :ref:`existential-variables`, which don't have names by default.
+     You can give a name to a goal by using :n:`refine ?[@ident]`.
 
-      Goals are just existential variables and existential variables do not
-      get a name by default. You can give a name to a goal by using :n:`refine ?[@ident]`.
-      You may also wrap this in an Ltac-definition like:
+   .. example:: Working with named goals
 
       .. coqtop:: in
 
-         Ltac name_goal name := refine ?[name].
-
-   .. seealso:: :ref:`existential-variables`
-
-   .. example::
-
-      This first example uses the Ltac definition above, and the named goals
-      only serve for documentation.
+         Ltac name_goal name := refine ?[name].  (* for convenience *)
+         Set Printing Goal Names.  (* show goal names, e.g. "(?base)" and "(?step)" *)
 
       .. coqtop:: all
 
          Goal forall n, n + 0 = n.
          Proof.
          induction n; [ name_goal base | name_goal step ].
+         (* focus on the goal named "base" *)
          [base]: {
-
-      .. coqtop:: all
-
-         reflexivity.
+           reflexivity.
 
       .. coqtop:: in
 
          }
 
-      .. coqtop:: all
-
-         [step]: {
-
-      .. coqtop:: all
-
-         simpl.
-         f_equal.
-         assumption.
-         }
-         Qed.
+      .. todo PR: Following seems unrealistic, maybe should be a separate example?
```

Comment:
This simplified case is unrealistic, but it can happen that existential variables remain unsolved when all goals are solved and this is the most realistic example I could find that is also sufficiently simple.